### PR TITLE
migrate to vitest

### PR DESCRIPTION
**Pull request by B2o5T**

### CODE REVIEW

**Comment by B2o5T** [comment]
I could fix by myself but need help to run integrations tests locally

**Comment by enisdenjo** [comment]
@B2o5T this is how you run the integration tests locally:

1. `export DOCKER_REGISTRY="local-graphql-hive-"`
1. `export DOCKER_TAG="testing"`
1. `pnpm i`
1. `pnpm build`
1. `pnpm docker:build`
1. `docker compose -f docker-compose.community.yml -f ./integration-tests/docker-compose.integration.yaml --env-file ./integration-tests/.env up -d --wait`
1. `pnpm --filter integration-tests test:integration`

**Comment by kamilkisiela** [comment]
If we want to use vitest in all of our projects then Hive is part of it, so I'm okay with migrating to vitest here :)

File: integration-tests/tests/api/artifacts-cdn.spec.ts [hunk 1]
```diff
@@ -313,7 +316,7 @@ function runArtifactsCDNTests(
         `artifact/${target.id}/services`,
       );
       expect(artifactContents.body).toMatchInlineSnapshot(
-        `"[{"name":"ping","sdl":"type Query { ping: String }","url":"ping.com"}]"`,
+        '"[{\\"name\\":\\"ping\\",\\"sdl\\":\\"type Query { ping: String }\\",\\"url\\":\\"ping.com\\"}]"',
```
Comment (enisdenjo) [inline integration-tests/tests/api/artifacts-cdn.spec.ts:319]: The former was easier to read, do we need to use single quotes?

**Review by enisdenjo — APPROVED** [review]

File: integration-tests/tests/api/artifacts-cdn.spec.ts (hunk 1 above)
Comment (B2o5T) [reply integration-tests/tests/api/artifacts-cdn.spec.ts:319]: It was auto update snapshots, I can add serializer to remove redundant escapes like in graphql-eslint https://github.com/B2o5T/graphql-eslint/blob/cfbb649f89b76673b8fdd718d3c227bada19a4a6/serializer.js

**Comment by kamilkisiela** [comment]
@B2o5T a rebase and we're ready to go

**Pull request merged**